### PR TITLE
feat: Update InfluxQL end-to-end test output

### LINKED ISSUE #7145: InfluxQL-IOx: end-to-end query tests should include metadata in expected output
@alamb [suggested a great idea](https://github.com/influxdata/influxdb_iox/pull/7109/files/f6c07f8637eccf295eb2b09578d248694bdc9589#r1126281032) to include the additional InfluxQL metadata in the expected output, shown as `Metadata` in the following example:

```text
-- InfluxQL: SELECT cpu, usage_idle FROM cpu;
Metadata: tag_columns: [2]
+------------------+----------------------+-----------+------------+
| iox::measurement | time                 | cpu       | usage_idle |
+------------------+----------------------+-----------+------------+
| cpu              | 2022-10-31T02:00:00Z | cpu-total | 2.98       |
| cpu              | 2022-10-31T02:00:00Z | cpu0      | 0.98       |
| cpu              | 2022-10-31T02:00:00Z | cpu1      | 1.98       |
| cpu              | 2022-10-31T02:00:10Z | cpu-total | 2.99       |
| cpu              | 2022-10-31T02:00:10Z | cpu0      | 0.99       |
| cpu              | 2022-10-31T02:00:10Z | cpu1      | 1.99       |
+------------------+----------------------+-----------+------------+
```

Requires:

- https://github.com/apache/arrow-rs/issues/3779

### PR DESCRIPTION
**Pull request by stuartcarnie**
Closes #7145

This PR includes the following changes:

* Updates the expected result output for InfluxQL tests to a format that more closely matching OSS. 
* Fixes an error uncovered in the `write_columnar` when tag columns are `NULL`
* Removed the explicit sorting of test output, which was masking an issue with InfluxQL result sorting.
* Added InfluxQlMetadata to `EXPLAIN` output, so that it may be used in end-to-end tests and the command line when using the `write_columnar` output.
* Added `IOX_COMPARE: no_borders` option, to remove borders from tabular output.
    * This avoids variable width issues with `EXPLAIN` output, which tends to vary from local to CI and cause flaky test failures.
    * ✅ it has resolved CI flakiness

> **Note**
>
> These changes also uncovered a sorting bug via #7513 and https://github.com/apache/arrow-datafusion/issues/5970
>
> I've re-added the explicit sorting to specific results until the root cause of #7513 is understood

### CODE REVIEW

File: influxdb_iox/tests/query_tests2/cases/in/issue_6112.influxql.expected [hunk 1]
```diff
@@ -1,1327 +1,2333 @@
...
++---------------------+-------+------+--------+
+name: m0
+tags: tag0=val02
++---------------------+-------+------+--------+
+| time                | count | sum  | stddev |
++---------------------+-------+------+--------+
+| 1970-01-01T00:00:00 | 1     | 10.4 |        |
++---------------------+-------+------+--------+
+
+-- InfluxQL: EXPLAIN SELECT COUNT(f64), SUM(f64), stddev(f64) FROM m0, m1 GROUP BY tag0;
```
Comment (stuartcarnie) [inline influxdb_iox/tests/query_tests2/cases/in/issue_6112.influxql.expected:886]: @crepererum this plan appears to be correctly sorting the output, however, there results are not sorting consistently, which means the subsequent test is flakey.

File: influxdb_iox/tests/query_tests2/cases/in/issue_6112.influxql.expected [hunk 2]
```diff
@@ -1,1327 +1,2333 @@
...
+|             RepartitionExec: partitioning=Hash([Column { name: "tag0", index: 0 }], 4), input_partitions=4                                                                      |
+|               RepartitionExec: partitioning=RoundRobinBatch(4), input_partitions=1                                                                                              |
+|                 AggregateExec: mode=Partial, gby=[tag0@1 as tag0], aggr=[COUNT(m1.f64), SUM(m1.f64), STDDEV(m1.f64)]                                                            |
+|                   ParquetExec: limit=None, partitions={1 group: [[1/1/1/1/00000000-0000-0000-0000-000000000001.parquet]]}, output_ordering=[tag0@1 ASC], projection=[f64, tag0] |
+|                                                                                                                                                                                 |
++---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+
 -- InfluxQL: SELECT COUNT(f64), SUM(f64), stddev(f64) FROM m0, m1 GROUP BY tag0;
```
Comment (stuartcarnie) [inline influxdb_iox/tests/query_tests2/cases/in/issue_6112.influxql.expected:925]: @crepererum specifically, this test is failing to sort correctly some of the time

File: influxdb_iox/tests/query_tests2/cases/in/issue_6112.influxql.expected (hunk 1 above)
Comment (crepererum) [reply influxdb_iox/tests/query_tests2/cases/in/issue_6112.influxql.expected:824]: We could add some "normalize" attribute like we have for SQL.

Comment (stuartcarnie) [reply influxdb_iox/tests/query_tests2/cases/in/issue_6112.influxql.expected:824]: 👍🏻 I have added that support to this PR and noted that it should be removed once #7513 is fixed.

I just need to figure out why CI is formatting the `EXPLAIN` output table wider that locally 🤦🏻

Comment (stuartcarnie) [reply influxdb_iox/tests/query_tests2/cases/in/issue_6112.influxql.expected:824]: @crepererum fixed the formatting issues for `EXPLAIN`

**Review by crepererum — APPROVED** [review]
That's awesome, thank you very much!

I'm taking the liberty to merge this so I can merge the other InfluxQL PRs w/o ending up in a massive conflict hell (also because it helps w/ the other PRs).

Feel free to file a follow-up PR if there's anything else to address.

**Pull request merged**